agent/dockerclient/dockerauth/ecr.go

```diff
@@ -52,7 +55,8 @@ const (
 
 // String formats the cachKey as a string
 func (key *cacheKey) String() string {
-	return fmt.Sprintf("%s-%s-%s-%s", key.roleARN, key.region, key.registryID, key.endpointOverride)
+	return fmt.Sprintf("%s%s-%s-%s-%s",
```

Comment:
Should we have a hyphen between credentialsID and roleARN?

Reply:
This way we don't change the cache key if there's no credentials ID to add (for example when we dont use task execution role). I dont _think_ this cache is stored to disk so probably doesn't matter, but seemed safer to do this without the dash anyways.

Reply:
Can the cache key be just the `credentialsID`? If the credentialsID is the same, there's a cache hit, else we get credentials from ECR.

Reply:
It can be, although we only get credentialsID when we're using the task execution role, we dont get them when we use the EC2 instance role, so we'd need to keep the rest of the fields anyways. Using only credentials ID for when we have it would fork the behavior a bit more than I wanted to change it.